lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -106,13 +113,13 @@ def execute(self) -> int:
 
                     host = path.name
                     tool_info[group][host] = {}
-                    # Check to see if the tool is in any of the hosts.
+                    tool_info[group][host]["label"] = None
+                    tool_info[group][host]["tools"] = {}
+
+                    # Check if the tool is in any of the hosts.
```

Comment:
The code at lines 101-122 seems an awful lot like the code at lines 63-83.  There are some differences which look like they shouldn't be there (like one uses globbing while the other uses `iterdir()`), and the remaining structure looks so close that I wonder if the two shouldn't be combined (with the `name` conditional used at the deep level instead at the top).

Reply:
I'll see what I can do.

Reply:
I've rearranged the code along the lines you suggested.